src/institution/dtos/institution.create.dto.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  Exclude,
+  Expose,
+  Transform,
+  TransformFnParams,
+} from 'class-transformer';
+import { IsOptional } from 'class-validator';
+import * as sanitizeHtml from 'sanitize-html';
+
+@Exclude()
+export class InstitutionCreateDto {
+  @Expose()
+  @Transform((params: TransformFnParams) => sanitizeHtml(params.value))
+  name: string;
+  @Transform((params: TransformFnParams) => sanitizeHtml(params.value))
+  abbreviation?: string;
+  @IsOptional()
+  @Transform((params: TransformFnParams) => sanitizeHtml(params.value))
```

Comment:
An html string is not expected here, maybe instead of sanitizing it we can reject the request if html tags are found, what do you think?

Reply:
Another validator like
```typescript
import { IsNotEmpty, Validate } from 'class-validator';
import { Transform } from 'class-transformer';
import { ValidatorConstraint, ValidatorConstraintInterface, ValidationArguments } from 'class-validator';

// Custom validator to check if the input is not an HTML document
@ValidatorConstraint({ name: 'isNotHtmlDocument', async: false })
class IsNotHtml implements ValidatorConstraintInterface {
  validate(text: string, args: ValidationArguments) {
    const htmlTagRegex = /<\/?[a-z][\s\S]*>/i;
    return !htmlTagRegex.test(text);
  }

  defaultMessage(args: ValidationArguments) {
    return 'Text should not be an HTML document';
  }
}

export class InstitutionCreateDto {
  @IsNotEmpty()
  @Validate(IsNotHtml)
  someProperty: string;
}
``